components/experimental/src/units/ratio.rs

```diff
@@ -193,26 +211,194 @@ impl From<u32> for IcuRatio {
-        let denominator = parts.next();
+            Ok(numerator / denominator)
+        }
+
+        /// Parses a decimal number from a string input.
+        /// Accepts input in various decimal formats, including:
+        /// - Whole numbers: "1"
+        /// - Decimal numbers: "1.5"
+        /// An empty string input is interpreted as "0".
+        /// Note: Fractional inputs are not supported in this context.
+        fn parse_decimal(decimal: &str) -> Result<IcuRatio, RatioFromStrError> {
+            let mut dot_parts = decimal.split('.');
+            let integer_part = dot_parts.next();
+            let decimal_part = dot_parts.next();
+            if dot_parts.next().is_some() {
+                return Err(RatioFromStrError::MultipleDecimalPoints);
+            }
+
+            let integer_part = match integer_part {
+                None | Some("") => IcuRatio::zero(),
+                Some(integer_part) => IcuRatio(
+                    BigInt::from_str(integer_part)
+                        .map_err(RatioFromStrError::ParsingBigIntError)?
+                        .into(),
+                ),
+            };
+
+            let decimal_part = match decimal_part {
+                None | Some("") => return Ok(integer_part),
+                Some(decimal_part) => {
+                    let decimal_power = decimal_part.len() as i32;
+                    let decimal_part = IcuRatio(
+                        BigInt::from_str(decimal_part)
+                            .map_err(RatioFromStrError::ParsingBigIntError)?
+                            .into(),
+                    );
+
+                    decimal_part / IcuRatio::ten().pow(decimal_power) // Divide by 10^decimal_power
+                }
+            };
+
+            Ok(integer_part + decimal_part)
+        }
+
+        let number_str = number_str.replace(',', ""); // Remove commas for parsing
+        if number_str.is_empty() {
+            return Ok(IcuRatio::zero());
+        }
+
+        let mut parts = number_str.split(|c| c == 'e' || c == 'E');
+        let significand = parts.next();
+        let exponent = parts.next();
         if parts.next().is_some() {
-            return Err(IcuRatioError::InvalidRatioString);
+            return Err(RatioFromStrError::MultipleScientificNotations);
         }
 
-        let numerator = match numerator {
-            Some(num_str) => BigInt::from_str(num_str).map_err(IcuRatioError::BigIntParseError)?,
-            None => BigInt::zero(),
+        let significand = match significand {
+            Some(significand) => parse_fraction(significand)?,
+            None => IcuRatio::one(),
         };
 
-        let denominator = match denominator {
-            Some(den_str) => BigInt::from_str(den_str).map_err(IcuRatioError::BigIntParseError)?,
-            None => BigInt::one(),
+        let exponent = match exponent {
+            Some(exponent) => {
+                let exponent = exponent
+                    .parse::<i32>()
```

Comment:
is `i32` always enough?

Reply:
yes, if it is more, it will return an error

Reply:
this is the exponent part